cdc/cdc.cc

```diff
@@ -321,16 +321,45 @@ class transformer final {
             // should be update or deletion
             int batch_no = 0;
             for (const rows_entry& r : p.clustered_rows()) {
-                auto log_ck = set_pk_columns(m.key(), batch_no, res);
                 auto ck_value = r.key().explode(*_schema);
+
+                std::optional<clustering_key> pikey;
+                const cql3::untyped_result_set_row * pirow = nullptr;
+
+                if (rs) {
+                    for (auto& utr : *rs) {
```

Comment:
I'm very sad about the quadratic complexity of this code. Would it be possible to depend on the fact that both p.clustered_rows() and rs are sorted and make it linear?

Reply:
Not for this PR though. I will do it as a follow up.